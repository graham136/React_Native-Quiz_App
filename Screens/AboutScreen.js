import * as React from 'react';
import { StyleSheet, Text, View, Button } from 'react-native';
import { ScrollView } from 'react-native-gesture-handler';

// About screen. Used to display info on the project/
const AboutScreen= ({navigation}) => {
  return (
    <View style={styles.container}>
      <ScrollView style={styles.container} contentContainerStyle={styles.contentContainer}>
        <View>
          <Text style = {styles.headerText}>Mathematics</Text>
          <Text style = {styles.headerText}>Grade 3</Text>
          <Text style = {styles.paragraphText}>App was written in React Native using Expo and Visual Studio Code</Text>         
          <Button title='Exit' onPress={() => {navigation.navigate('Home')}}/> 
        </View>
      </ScrollView>       
    </View>
  );
}

AboutScreen.navigationOptions = {
  header: null,
};

export default AboutScreen;

const styles = StyleSheet.create({
    container: {
      flex: 1,
      backgroundColor: '#fff',
    },
    contentContainer: {
      paddingTop: 30,
      marginVertical: 10,   
    },
    headerText: {
      fontSize: 30,
      color: 'dodgerblue',
      textAlign: 'center',
    }, 
  });
  