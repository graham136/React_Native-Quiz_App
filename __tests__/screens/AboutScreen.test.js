import React from 'react';
import renderer from 'react-test-renderer';

import AboutScreen from '../../Screens/AboutScreen';

describe('<AboutScreen />', () => {
  
  // Snaphot test
  it('renders correctly', async () => {
    const tree = renderer.create(<AboutScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});