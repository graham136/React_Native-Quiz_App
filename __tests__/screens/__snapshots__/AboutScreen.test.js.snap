// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AboutScreen /> renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    collapsable={false}
    contentContainerStyle={
      Object {
        "marginVertical": 10,
        "paddingTop": 30,
      }
    }
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    style={
      Object {
        "backgroundColor": "#fff",
        "flex": 1,
      }
    }
  >
    <View>
      <View>
        <Text
          style={
            Object {
              "color": "dodgerblue",
              "fontSize": 30,
              "textAlign": "center",
            }
          }
        >
          Mathematics
        </Text>
        <Text
          style={
            Object {
              "color": "dodgerblue",
              "fontSize": 30,
              "textAlign": "center",
            }
          }
        >
          Grade 3
        </Text>
        <Text>
          App was written in React Native using Expo and Visual Studio Code
        </Text>
        <View
          accessibilityRole="button"
          accessibilityState={Object {}}
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {},
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#007AFF",
                    "fontSize": 18,
                    "margin": 8,
                    "textAlign": "center",
                  },
                ]
              }
            >
              Exit
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;

exports[`<App /> renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    collapsable={false}
    contentContainerStyle={
      Object {
        "marginVertical": 10,
        "paddingTop": 30,
      }
    }
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    style={
      Object {
        "backgroundColor": "#fff",
        "flex": 1,
      }
    }
  >
    <View>
      <View>
        <Text
          style={
            Object {
              "color": "dodgerblue",
              "fontSize": 30,
              "textAlign": "center",
            }
          }
        >
          Mathematics
        </Text>
        <Text
          style={
            Object {
              "color": "dodgerblue",
              "fontSize": 30,
              "textAlign": "center",
            }
          }
        >
          Grade 3
        </Text>
        <Text>
          App was written in React Native using Expo and Visual Studio Code
        </Text>
        <View
          accessibilityRole="button"
          accessibilityState={Object {}}
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {},
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#007AFF",
                    "fontSize": 18,
                    "margin": 8,
                    "textAlign": "center",
                  },
                ]
              }
            >
              Exit
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
